evals/completion_fns/cot.py

```diff
@@ -0,0 +1,68 @@
+"""
+Extending OpenAI Models with Chain-of-Thought Inner Monologue
+"""
+from evals.prompt.base import ChatCompletionPrompt
+from evals.record import record_sampling
+from evals.registry import Registry
+
+DEFAULT_COT_TEMPLATE = "\nBefore answering, reason in a step-by-step manner as to get the right answer, then conclude with the answer."
+DEFAULT_EXTRACT_ANSWER_TEMPLATE = (
+    "\nGiven the above reasoning, the answer in the format requested by the question is:"
+)
+
+
+class ChainOfThoughtCompletionResult:
+    def __init__(self, response) -> None:
+        self.response = response
+
+    def get_completions(self) -> list[str]:
+        return [self.response.strip()]
+
+
+class ChainOfThoughtCompletionFn:
+    def __init__(
+        self,
+        cot_template=DEFAULT_COT_TEMPLATE,
+        extract_answer_template=DEFAULT_EXTRACT_ANSWER_TEMPLATE,
+        cot_completion_fn=None,
+        extract_completion_fn=None,
+        registry_path=None,
+        extra_options=None,
+        **kwargs
+    ) -> None:
+
+        self.extra_options = extra_options
+
+        registry = Registry()
```

Comment:
Ideally this would be the same instance that is created by `oaieval` rather than having to reconstruct it here. One way to do it could be to pass `registry` through to the `CompletionFn` constructor as a `kwarg` via `Registry.make_completion_fn`. But then it seems we would have to rely on the correct naming which seems not great.

Any other ideas?